src/main/java/ru/testtask/service/WebSocketEventListener.java

```diff
@@ -0,0 +1,43 @@
+package ru.testtask.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.messaging.simp.SimpMessageSendingOperations;
+import org.springframework.messaging.simp.stomp.StompHeaderAccessor;
+import org.springframework.stereotype.Component;
+import org.springframework.web.socket.messaging.SessionConnectedEvent;
+import org.springframework.web.socket.messaging.SessionDisconnectEvent;
+import ru.testtask.model.ChatMessage;
+
+import static java.lang.String.format;
+
+@Slf4j
+@Component
+public class WebSocketEventListener {
+
+    @Autowired
+    private SimpMessageSendingOperations messagingTemplate;
+
+    @EventListener
+    public void handleWebSocketConnectListener(SessionConnectedEvent event) {
+        log.info("Received a new web socket connection");
+    }
+
+    @EventListener
+    public void handleWebSocketDisconnectListener(SessionDisconnectEvent event) {
+        StompHeaderAccessor headerAccessor = StompHeaderAccessor.wrap(event.getMessage());
+
+        String username = (String) headerAccessor.getSessionAttributes().get("username");
+        String roomId = (String) headerAccessor.getSessionAttributes().get("room_id");
+        if (username != null) {
+            log.info("User Disconnected: " + username);
+
+            ChatMessage chatMessage = new ChatMessage();
+            chatMessage.setType(ChatMessage.MessageType.LEAVE);
+            chatMessage.setSender(username);
+
+            messagingTemplate.convertAndSend(format("/channel/%s", roomId), chatMessage);
```

Comment:
I see many similar code, like this: `messagingTemplate.convertAndSend(format("/channel/%s", roomId), ...`

Can you please think about new method like: 
`MessageService#sendMessageToChat(MessageType type, String content, String senderName)`

Or may be it should be several overloaded methods?